todo_app/app.py

```diff
@@ -3,26 +3,99 @@
 import os
 import requests
 from todo_app.mongo_actions import MongoActions
+from flask_login import LoginManager, login_required, login_user, UserMixin, current_user
+from oauthlib.oauth2 import WebApplicationClient
+from functools import wraps
+
+class User(UserMixin):
+    def __init__(self, userId):
+        self.id = userId
+        if userId == 'jainsushma':
+            self.role = "writer"
+        else: 
+            self.role = "reader"
 
 def create_app():
     app = Flask(__name__)
+    app.config.from_object('todo_app.flask_config.Config')
+    login_manager = LoginManager()
+    login_manager.init_app(app)
+
+    # Values setup during registration
+    clientId = os.getenv('CLIENTID')
+    clientSecret = os.getenv('CLIENTSECRET')
+    loginDisabled = app.config.get('LOGIN_DISABLED')
+    githubClient = WebApplicationClient(clientId)
+
+    @app.route('/login/callback')
+    def call_back():
+        callbackCode = request.args.get("code")
+        tokenUrl, headers, body = githubClient.prepare_token_request("https://github.com/login/oauth/access_token", code=callbackCode)
+        headers["Accept"] = "application/json"
+        r = requests.post(
+            tokenUrl,
+            headers=headers,
+            data=body,
+            auth=(clientId, clientSecret),
+        )
+        params=githubClient.parse_request_body_response(r.text)
+        print(params['access_token'])
```

Comment:
This is a sensitive token, so while it's very helpful for debugging we should remove it from our logs afterwards!

Reply:
Done!